### PR TITLE
Use `datetime-local` instead of datetime picker in event settings

### PR DESCRIPTION
**Pull request by SudharakaP**
Resolve https://github.com/opencollective/opencollective/issues/4865

event-creation

### CODE REVIEW

**Comment by Betree** [comment]
Do we still need `react-datetime`?

**Comment by Betree** [comment]
I get this when I try to edit the start date for any event, with any value

image

**Comment by znarf** [comment]
On my computer it works, but the end result doesn't match the specs. The stored date is not the date passed. Certainly related to Ben's issue.

**The date you see in the PostgresSQL row should be the one the user inputed and originally passed.**

**Comment by znarf** [comment]
I added a commit which seems to take care of this for the Edit part.

**Comment by znarf** [comment]
Added more commits because the behavior was not making much sense.

File: components/EditEventForm.js [hunk 1]
```diff
@@ -73,23 +74,10 @@ class EditEventForm extends React.Component {
...
-        } else {
-          // https://github.com/opencollective/opencollective/issues/1232
-          const endsAtDate = new Date(endsAt);
-          newEndDate = new Date(value);
-          newEndDate.setHours(endsAtDate.getHours());
-          newEndDate.setMinutes(endsAtDate.getMinutes());
-        }
-        value = newEndDate.toString();
-        event['endsAt'] = value;
-      }
```
Comment (znarf) [inline components/EditEventForm.js:92]: Just leave endsAt alone and no magic crazyness.

File: components/EditEventForm.js [hunk 2]
```diff
@@ -120,9 +116,9 @@ class EditEventForm extends React.Component {
 
     const isNew = !(event && event.id);
     const submitBtnLabel = loading ? 'loading' : isNew ? 'Create Event' : 'Save';
-    const defaultStartsAt = new Date();
-    defaultStartsAt.setHours(19);
-    defaultStartsAt.setMinutes(0);
+
+    const defaultStartsAt = dayjs().utc().set('hour', 19).set('minute', 0);
+    const defaultEndsAt = dayjs().utc().set('hour', 22).set('minute', 0);
```
Comment (znarf) [inline components/EditEventForm.js:121]: Use dayjs and utcise.

File: components/edit-collective/Form.js [hunk 3]
```diff
@@ -327,7 +328,7 @@ class EditCollectiveForm extends React.Component {
       } else if (fieldname === 'application.message') {
         set(collective, 'settings.applyMessage', value);
       } else if (fieldname === 'startsAt' && collective.type === EVENT) {
-        collective[fieldname] = value;
+        collective[fieldname] = dayjs(value).format('YYYY-MM-DD HH:mm:ss+00');
```
Comment (znarf) [inline components/edit-collective/Form.js:331]: This is the format expected by the GraphQL API v1 (Postgres)

File: components/edit-collective/Form.js [hunk 4]
```diff
@@ -649,10 +652,14 @@ class EditCollectiveForm extends React.Component {
...
+
     const type = collective.type.toLowerCase();
-    defaultStartsAt.setHours(19);
-    defaultStartsAt.setMinutes(0);
+
+    const defaultStartsAt = dayjs().utc().set('hour', 19).set('minute', 0);
+    const defaultEndsAt = dayjs(collective.startsAt || defaultStartsAt)
+      .utc()
+      .set('hour', 22)
+      .set('minute', 0);
```
Comment (znarf) [inline components/edit-collective/Form.js:662]: Use dayjs

**Comment by SudharakaP** [comment]
> Do we still need `react-datetime`?

@Betree : I don't think so, but I need to carefully check this one. So I've created a separate issue so that we can remove this; https://github.com/opencollective/opencollective/issues/5010

**Comment by SudharakaP** [comment]
@Betree @znarf : Thanks for the corrections. I've added some small things here; 

1) In our old date time picker it automatically had support for minimum date time (cannot pick a date before today). I've added this to the `datetime-local` as well by specifying the `min` attribute. 

2) Depending on the browser the `datetime-local` input might allow the user to specify nothing (Firefox in particular). And if they do that we probably need to have a nicer error message rather than the one @Betree saw above. I've added this at, https://github.com/opencollective/opencollective-api/pull/6915

**Comment by SudharakaP** [comment]
Seems a e2e test is failing after the latest commit. Looking into this. 🤔

**Comment by SudharakaP** [comment]
> Seems a e2e test is failing after the latest commit. Looking into this. thinking

This is due to my API PR; https://github.com/opencollective/opencollective-api/pull/6915. This problem isn't specific to this issue so therefore I'll tackle this on a separate issue at, https://github.com/opencollective/opencollective/issues/5011

**Pull request merged**